types/jasmine/index.d.ts

```diff
@@ -204,13 +204,19 @@ declare namespace jasmine {
             (ReadonlyArray<string> | { [methodName: string]: any }) :
             (ReadonlyArray<keyof T> | { [P in keyof T]?: T[P] extends Func ? ReturnType<T[P]> : any });
 
+    type SpyObjPropertyNames<T = undefined> =
+        T extends undefined ?
+                (ReadonlyArray<string> | { [propertyName: string]: any }) :
+                (ReadonlyArray<keyof T>);
+
     /**
      * Configuration that can be used when configuring Jasmine via {@link jasmine.Env.configure}
      */
     interface EnvConfiguration {
         random?: boolean;
         seed?: number;
         failFast?: boolean;
+        failSpecWithNoExpectations?: boolean;
```

Comment:
According to the [Configuration docs](https://jasmine.github.io/api/3.5/Configuration.html) there is also a `Promise` option since 3.5.0, please add this one as well